In [1]:
import os
for f in os.listdir('../fun/'): exec(open('../fun/'+f).read())
del f

# Load data.
load( '../out/d3-fight-level-transform.pkl' )
print( X.shape )

(4368, 165)


In [2]:
# train test split.
from sklearn.model_selection import train_test_split
X_train , X_test, y_train, y_test = train_test_split(
    X, y, 
    random_state = 718,
    test_size = 0.25
)

In [3]:
# Fit decision tree.
from sklearn.tree import DecisionTreeClassifier
m = DecisionTreeClassifier( max_depth = 6, min_samples_split = .1, random_state = 716 )
m.fit( X_train, y_train )

# accuracy against train data (in-model).
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_train, m.predict(X_train) ) )

              precision    recall  f1-score   support

           0       0.59      0.41      0.48      1056
           1       0.76      0.87      0.81      2220

    accuracy                           0.72      3276
   macro avg       0.67      0.64      0.65      3276
weighted avg       0.70      0.72      0.70      3276



In [4]:
# accuracy against test data.
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_test, m.predict(X_test) ) )

              precision    recall  f1-score   support

           0       0.50      0.29      0.37       384
           1       0.69      0.84      0.76       708

    accuracy                           0.65      1092
   macro avg       0.59      0.57      0.56      1092
weighted avg       0.62      0.65      0.62      1092



f1-score in-model is higher than on test indicating **over-fitting**. 

Test-set precision is still not great at 69% (predicting all-wins give 60%).